plugins/modules/aci_cloud_ctx_profile.py

```diff
@@ -0,0 +1,234 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = r'''
+---
+module: aci_cloud_ctx_profile
+short_description: Manage Cloud Context Profile (cloud:CtxProfile)
+description:
+- Mo doc not defined in techpub!!!
+notes:
+- More information about the internal APIC class B(cloud:CtxProfile) from
+  L(the APIC Management Information Model reference,https://developer.cisco.com/docs/apic-mim-ref/).
+author:
+- Devarshi Shah (@devarshishah3)
+version_added: '2.7'
+options:
+  descr:
+    description:
+    - configuration item description.
+    type: str
+  name:
+    description:
+    - object name
+    type: str
+    aliases: [ cloud_context_profile ]
+  nameAlias:
+    description:
+    - Mo doc not defined in techpub!!!
+    type: str
+  type:
+    description:
+    - component type
+    choices: [ regular, shadow ]
+    type: str
+  tenant:
+    description:
+    - tenant name
+    type: str
+  primary_cidr:
+    description:
+    - cidr block range of primary cidr
+    type: str
+  vrf:
+    description:
+    - name of vrf to be managed
+    type: str
+  region:
+    description:
+    - name of region to be managed
+    type: str
+  vpn_gateway:
+    description:
+    - whether vpn gateway router is enabled or not
+    type: bool
+  state:
+    description:
+    - Use C(present) or C(absent) for adding or removing.
+    - Use C(query) for listing an object or multiple objects.
+    choices: [ absent, present, query ]
+    type: str
+    default: present
+
+extends_documentation_fragment:
+- cisco.aci.aci
+'''
+
+EXAMPLES = r'''
+- name: Add a new aci cloud ctx profile
+  aci_cloud_ctx_profile:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    tenant: tenant_1
+    name: cloud_ctx_profile
+    vrf: VRF1
+    region: us-west-1
+    primary_cidr: '10.0.10.1/16'
+    state: present
+  delegate_to: localhost
+
+- name: Remove an aci cloud ctx profile
+  aci_cloud_ctx_profile:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    tenant: tenant_1
+    name: cloud_ctx_profile
+    state: absent
+  delegate_to: localhost
+
+- name: Query aci cloud ctx profile
+  aci_cloud_ctx_profile:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    tenant: anstest
+    name: ctx_profile_1
+    state: query
+  delegate_to: localhost
+'''
```

Comment:
Are we able to query all the cloud context profiles? if yes, please add an example.